### PR TITLE
[TIMOB-24608] Android: respackage value in native module's manifest is ignored

### PR DESCRIPTION
**Pull request by sgtcoolguy**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-24608

**Description:**
We never consulted the module's `manifest` file's `respackage` value (in our new node-based module build). We need to look for a non-empty value and if it exists generate a `respackageinfo` file inside the module zip with the value. This is used to generated the R* java/class files for resources on Android.

### CODE REVIEW

**Comment by sgtcoolguy** [comment]
**For QE/FR**
Use a native android module like ti.facebook that contains resources (and not just a jar file). The manifest should have a `repackage` value.
Run `appc run -p android --build-only`
The resulting zip file in `android/dist` should have a `respackageinfo` file in the root of it, with the  manifest's `repackage` value.

File: android/cli/commands/_buildModule.js [hunk 1]
```diff
@@ -1474,6 +1474,11 @@ AndroidModuleBuilder.prototype.packageZip = function (next) {
 					}.bind(this));
 				}
 
+				// respackageinfo file
+				if (this.manifest.respackage) {
```
Comment (cb1kenobi) [inline android/cli/commands/_buildModule.js]: Why add a `respackage` key to the manifest instead of just checking if the `respackageinfo` file exists?

Comment (sgtcoolguy) [inline android/cli/commands/_buildModule.js]: I'm not sure I understand. The `respackage` value is not a boolean key, it's an optional manifest value that typically is a java package name - and it's set manually by the user. We look to see if it's set to anything and if so, generate a `respackageinfo` file in the module holding the value.

So we either need to copy it from he manifest to the file expected, or change our app build process that handles modules to not look for this file but instead consult the manifest.

Comment (cb1kenobi) [inline android/cli/commands/_buildModule.js]: Nevermind, I see what it's doing.

**Review by cb1kenobi — APPROVED** [review]

**Comment by lokeshchdhry** [comment]
FR Passed.

1. Downloaded repo for facebook module from master branch.
2. Selected the SDK with this fix in the CLI.
3. Ran `appc run -p android --build-only` from the android folder.
4. The resulting zip file in the `dist` folder has the correct `respackageinfo` file at the root with the manifest's `repackage` value.

Studio Ver:     4.9.0.201705302345
SDK Ver:        6.2.0 local build
OS Ver:         10.12.3
Xcode Ver:      Xcode 8.3.2
Appc NPM:       4.2.9
Appc CLI:       6.2.2
Ti CLI Ver:     5.0.14
Alloy Ver:      1.9.11
Node Ver:       6.10.1
Java Ver:       1.8.0_101

**Pull request merged**